app/src/androidTest/java/net/skyscanner/backpack/compose/appsearchmodal/BpkAppSearchModalTest.kt

```diff
@@ -0,0 +1,72 @@
+/**
+ * Backpack for Android - Skyscanner's Design System
+ *
+ * Copyright 2018 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.skyscanner.backpack.compose.appsearchmodal
+
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.test.isDialog
+import net.skyscanner.backpack.compose.BpkSnapshotTest
+import net.skyscanner.backpack.compose.theme.BpkTheme
+import net.skyscanner.backpack.demo.compose.AppSearchModalStoryContent
+import net.skyscanner.backpack.demo.compose.AppSearchModalStoryError
+import net.skyscanner.backpack.demo.compose.AppSearchModalStoryLoading
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.junit.runners.Parameterized
+
+@RunWith(Parameterized::class)
+class BpkAppSearchModalTest(private val permutation: Permutation) : BpkSnapshotTest(listOf(permutation)) {
+
+    @Test
+    fun default() = record {
+        permutation.composable()
+    }
+
+    @Test
+    fun on_dark() = record(background = { BpkTheme.colors.surfaceContrast }) {
```

Comment:
not sure what this is for? there's no separate on dark mode of the modal, so this shouldn't be needed. i think just keeping it simple without a permutation and having a `content` `error` and `loading` test is enough

Reply:
I added on dark just to make sure we check screenshot tests for the case when modal is dark, and be 100% it looks as intended 😅

Reply:
i don't think it's needed tbh - we already have the light/dark screenshots that ensure the background is updated/correct